### DISCUSSION TITLE
Pinout Images - Feedback wanted

### DISCUSSION THREAD

**Discussion by ObviousInRetrospect** [SpenceKonde/DxCore#335] [General]
I took up #264 and wrote some scripts that mostly automate the generation of pinout images from tables extracted from the datasheets as well as the atdf files. The positives of this approach are that it should be easier to get consistent accurate diagrams with less opportunity for human error. Unfortunately my visual design skills are limited. The images are here:

DD:
https://raw.githubusercontent.com/ObviousInRetrospect/DxCore/pinouts-with-db/megaavr/extras/AVR32DD14.svg
https://raw.githubusercontent.com/ObviousInRetrospect/DxCore/pinouts-with-db/megaavr/extras/AVR32DD20-QFN.svg
https://raw.githubusercontent.com/ObviousInRetrospect/DxCore/pinouts-with-db/megaavr/extras/AVR32DD20-10x2.svg
https://raw.githubusercontent.com/ObviousInRetrospect/DxCore/pinouts-with-db/megaavr/extras/AVR64DD28-14x2.svg
https://raw.githubusercontent.com/ObviousInRetrospect/DxCore/pinouts-with-db/megaavr/extras/AVR64DD28-QFN.svg
https://raw.githubusercontent.com/ObviousInRetrospect/DxCore/pinouts-with-db/megaavr/extras/AVR64DD32.svg

For DD these are currently all we have. But since DD support isn't quite there yet I figured I would also generate this style of diagram for all mcus supported by DxCore and megaTinyCore. While I used the 64kb ones as the datasheets were newer they are all the same pinout.

DB:
https://raw.githubusercontent.com/ObviousInRetrospect/DxCore/pinouts-with-db/megaavr/extras/AVR64DB28.svg
https://raw.githubusercontent.com/ObviousInRetrospect/DxCore/pinouts-with-db/megaavr/extras/AVR64DB32.svg
https://raw.githubusercontent.com/ObviousInRetrospect/DxCore/pinouts-with-db/megaavr/extras/AVR64DB48.svg
https://raw.githubusercontent.com/ObviousInRetrospect/DxCore/pinouts-with-db/megaavr/extras/AVR64DB64.svg

DA: (less verification than the others)
https://raw.githubusercontent.com/ObviousInRetrospect/DxCore/pinouts-with-db/megaavr/extras/AVR64DA28.svg
https://raw.githubusercontent.com/ObviousInRetrospect/DxCore/pinouts-with-db/megaavr/extras/AVR64DA32.svg
https://raw.githubusercontent.com/ObviousInRetrospect/DxCore/pinouts-with-db/megaavr/extras/AVR64DA48.svg
https://raw.githubusercontent.com/ObviousInRetrospect/DxCore/pinouts-with-db/megaavr/extras/AVR64DA64.svg

I'd love any feedback around what is clear, unclear, and how these can be made more useful and prettier. The color scheme is roughly greens: analog, blues: serial, reds: timers, grey: special, oranges other

Known Issues:
* 64-pin and 48-pin are probably too small and unoptimized
* DB legend has some cruft at the end
* Proofreading done but not exhaustive. Omissions are by far the most likely error.

I'm also curious whether people would strongly prefer that square chips be rendered square like the current diagrams or if people are ok with a rectangular rendering that is equivalent to mounting the square chip on a breakout diagonally where the first half goes on the left and the second half on the right which has the advantage of keeping all the text oriented the same way.

If anyone wants to dig in more or test changes, https://github.com/ObviousInRetrospect/DxCore/tree/pinouts-with-db/megaavr/extras/GenPinoutSVG contains the inputs to https://github.com/ObviousInRetrospect/GenPinoutSVG (a fork with a trivial change to support multiple boxes per column).

[edit: repointed links at new files generated off atpack atdf data and at the raw view which is easier to read]

**Comment by ObviousInRetrospect** [comment]
(looking into what can be extracted from the atdf xml which might have been the right place to start)

**Comment by matou78** [comment]
They look nice, but i think it would be better to have the real form of the chip on the image,
I love the images of chips from this core as it make it easier for my shitty brain to see what is where on the real chip .

**Reply by ObviousInRetrospect** [reply]
see https://github.com/SpenceKonde/megaTinyCore/discussions/786 for a longer discussion and a bunch of updates (both cores covered)

see https://github.com/ObviousInRetrospect/avr_pinout_diagrams/tree/main/svg for the latest proposed diagrams

I assume you are talking about the shape of the chip, that has been addressed: https://raw.githubusercontent.com/ObviousInRetrospect/avr_pinout_diagrams/main/svg/AVR64DB32.svg

or the bleeding edge more compact format (still in development):

![AVR64DB32](https://user-images.githubusercontent.com/42229858/188675533-d92cd055-03aa-4462-894d-5a95522b0d02.svg)

[edits: the o in the wrong location was bugging me, also grey looks better]
